### PR TITLE
Improved validation error message when env.valueFrom contains no (or …

### PR DESCRIPTION
**Pull request by luksa**
**What this PR does / why we need it**:
A misleading error message is shown if the user mistypes (or forgets to specify) a field under env.valueFrom. This is the error message: "may not have more than one field specified at a time". But there is only one (misspelled) field specified.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```
Improved error message for missing/misspelled field under env.valueFrom
```

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36449)

### CODE REVIEW

**Comment by dims** [comment]
@k8s-bot ok to test

File: pkg/api/validation/validation.go [hunk 1]
```diff
@@ -1350,11 +1350,13 @@ func validateEnvVarValueFrom(ev api.EnvVar, fldPath *field.Path) field.ErrorList
 		allErrs = append(allErrs, validateSecretKeySelector(ev.ValueFrom.SecretKeyRef, fldPath.Child("secretKeyRef"))...)
 	}
 
-	if len(ev.Value) != 0 {
+	if numSources == 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath, "", "must specify one of: fieldRef, resourceFieldRef, configMapKeyRef or secretKeyRef"))
```
Comment (thockin) [inline pkg/api/validation/validation.go]: field names should be backtick-quoted (see `value` in the next block)

**Comment by thockin** [comment]
Thanks for the PR.  You need to sign the LinuxFoundation CLA before this can be merged,

**Comment by luksa** [comment]
FYI: LinuxFoundation CLA signed two days ago. k8s-bot apparently doesn't recheck automatically. I guess someone has to trigger the recheck.

**Pull request merged**